test/grouping.jl

```diff
@@ -1256,4 +1276,131 @@ end
+    @test_throws KeyError gd[(:A, :X, 1)]
+    @test_throws KeyError gd[(b=:X, a=:D)]
+end
+
+@testset "GroupKey and GroupKeys" begin
+    df = DataFrame(a = repeat([:A, :B, missing], outer=4), b = repeat([:X, :Y], inner=6), c = 1:12)
+    cols = [:a, :b]
+    colstup = Tuple(cols)
+    gd = groupby_checked(df, cols)
+    gdkeys = keys(gd)
+
+    expected =
+        [(a=:A, b=:X), (a=:B, b=:X), (a=missing, b=:X), (a=:A, b=:Y), (a=:B, b=:Y), (a=missing, b=:Y)]
+
+    # Check AbstractVector behavior
+    @test length(gdkeys) == length(expected)
+    @test size(gdkeys) == size(expected)
+    @test eltype(gdkeys) == DataFrames.GroupKey{typeof(gd)}
+
+    # Test each key
+    cnt = 0
+    for (i, key) in enumerate(gdkeys)
+        cnt += 1
+        nt = expected[i]
+
+        # Check iteration vs indexing of GroupKeys
+        @test key == gdkeys[i]
+
+        # Basic methods
+        @test parent(key) === gd
+        @test length(key) == length(cols)
+        @test names(key) == cols
+        @test keys(key) == colstup
+        @test propertynames(key) == colstup
+        @test propertynames(key, true) == colstup
+        @test values(key) ≅ values(nt)
+
+        # (Named)Tuple conversion
+        @test Tuple(key) ≅ values(nt)
+        @test NamedTuple(key) ≅ nt
+
+        # Iteration
+        @test collect(key) ≅ collect(nt)
+
+        # Integer/symbol indexing, getproperty of key
+        for (j, n) in enumerate(cols)
+            @test key[j] ≅ nt[j]
+            @test key[n] ≅ nt[j]
+            @test getproperty(key, n) ≅ nt[j]
+        end
+
+        # Invalid key/property of key
+        @test_throws KeyError key[:foo]
+        @test_throws ArgumentError key.foo
+
+        # Using key to index GroupedDataFrame
+        @test gd[key] ≅ gd[i]
+    end
+
+    # Make sure we actually iterated over all of them
+    @test cnt == length(gd)
+
+    # Indexing using another GroupedDataFrame instance should fail
+    gd2 = groupby(df, cols, skipmissing=true)
+    gd3 = groupby(df, cols, skipmissing=true)
+    @test gd2 == gd3  # Use GDF's without missing so they compare equal
+    @test_throws ErrorException gd3[first(keys(gd2))]
+
+    # Key equality
+    @test collect(keys(gd)) == gdkeys  # These are new instances
+    @test all(Ref(gdkeys[1]) .!= gdkeys[2:end])
+    @test !any(collect(keys(gd2)) .== keys(gd3))  # Same values but different (but equal) parent
+end
+
+@testset "Parent DataFrame names changed" begin
+    df = DataFrame(a = repeat([:A, :B, missing], outer=4), b = repeat([:X, :Y], inner=6), c = 1:12)
+    gd = groupby_checked(df, [:a, :b])
+
+    @test names(gd) == names(df)
+    @test groupvars(gd) == [:a, :b]
+    @test map(NamedTuple, keys(gd)) ≅
+        [(a=:A, b=:X), (a=:B, b=:X), (a=missing, b=:X), (a=:A, b=:Y), (a=:B, b=:Y), (a=missing, b=:Y)]
+    @test gd[(a=:A, b=:X)] ≅ gd[1]
+    @test gd[keys(gd)[1]] ≅ gd[1]
+    @test NamedTuple(keys(gd)[1]) == (a=:A, b=:X)
+    @test keys(gd)[1].a == :A
+
+    names!(df, [:d, :e, :f])
```

Comment:
`names!` is deprecated - use `rename!`

Reply:
Why is this resolved? I do not see the change in the code.

Reply:
Sorry, I mark these off preemptively as I fix them in my local repo. Usually push the changes soon after but haven't yet this time.